src/nvim/msgpack_rpc/channel.c

```diff
@@ -312,6 +315,9 @@ bool channel_close(uint64_t id)
   }
 
   close_channel(channel);
+  if (!kv_size(channel->call_stack)) {
```

Comment:
Honestly, me too. I'd prefer just one single place where real freeing is done.
I just thought others would not like checking redundant conditions.
But if nobody opposes, I'll change it so that only one place encapsulates conditions to be able to free, and only one place calls channel_free.